third_party/silabs/SiWx917_sdk.gni

```diff
@@ -138,6 +143,26 @@ template("efr32_sdk") {
       "RS91X_BLE_ENABLE=1",
     ]
 
+    if (use_silabs_sw_versioning) {
+      formatted_version =
+          exec_script("${chip_root}/scripts/tools/silabs/format_sw_version.py",
```

Comment:
I think the guidance I got from gn build systems is that exec script should be generally banned inside build execution.

Also is this creating reproducible builds? It seems that it would re-build regardless of what changes in the repo. It seems preferable to send a version string into gn and let that be controlled by the user. Having the build system auto-pick a git version does not seem ideal.

Reply:
@andy31415 Responding to your few comments here.

exec_script is used more generally than only `pkg_config` and in core matter build, more notably to list the cluster sources in `chip_date_model.gni`. Not sure I understand why exec_script is generally banned. In makefiles, you have access to shell commands, isn't exec_script allowing similar capabilities?

> Also is this creating reproducible builds?
The version only changes if you change branch or commit SHA. We feel those informations are relevant to us and facilitate the debugging process when reviewing logs that are provided to us by clients. To me, the build time increase, if any of those info changes, is worth it. I still allowed the option to toggle it off with the build argument `use_silabs_sw_versioning=flase`

I can use the build argument idea to pass my string in the build system. In fact, it was my first idea, generating the version format in my build script and passing it on. It will lead to the same result, same pros and cons except that I have no control over what clients would use as a build script. They can implement their own or use gn gen/ ninja -C out directly and avoid what I deem important to our platform's sample debugging.

If you have a better solution, I am happy to change my implementation

Reply:
The chip_data_model.gni usage is/was wrong and should be corrected.

Reply:
I am fine with having a `version` variable and set that during build scripts - i.e. not auto-computed by gn.

Reply:
This will not work correctly in incremental builds because ninja cannot tell if the hash changes. If you want a version that works correctly you can pass it via args

gn gen --args="build_version=$hash"

Reply:
Based on our conversations, I reworked this PR. The string is formatted/determined at precompile time in the build scripts. The resultant string is passed to gn as a new arg.

I tested a few ways/locations to set the string to define for the Silabs examples. This selected option is the one that leads to the smallest recompilation/build time if the version string changes (branch change or new commit). 20 objects need to be recompiled such a scenario.